src/ProductConstructionService/SubscriptionTriggerer/SubscriptionTriggererConfiguration.cs

```diff
@@ -0,0 +1,107 @@
+
+    public static IServiceCollection RegisterServices(IServiceCollection services, ITelemetryChannel telemetryChannel)
+    {
+        RegisterLogging(services, telemetryChannel);
+
+        IConfiguration config = GetConfiguration();
+        services.AddSingleton(_ => config);
+
+        string databaseConnectionString = config.GetRequiredValue(DatabaseConnectionString)
+            .Replace(SqlConnectionStringUserIdPlaceholder, config[ManagedIdentityClientId]);
+
+        services.AddBuildAssetRegistry((provider, options) =>
+        {
+            options.UseSqlServerWithRetry(databaseConnectionString);
+        });
+
+        services.Configure<OperationManagerOptions>(o => { });
+        services.AddTransient<OperationManager>();
+
+        services.AddTransient<IHostEnvironment>(_ => new HostEnvironment());
+        services.AddTransient<DarcRemoteMemoryCache>();
+        services.AddTransient<IProcessManager>(sp => ActivatorUtilities.CreateInstance<ProcessManager>(sp, "git"));
+        services.AddTransient<IVersionDetailsParser, VersionDetailsParser>();
+        services.AddTransient<IBasicBarClient, SqlBarClient>();
+        services.AddTransient(_ => new QueueClient(
+            new Uri(config.GetRequiredValue(QueueConnectionString)),
+            new DefaultAzureCredential(new DefaultAzureCredentialOptions
+            {
+                ManagedIdentityClientId = config[ManagedIdentityClientId]
+            })));
+        services.AddKustoClientProvider("Kusto");
+
+        return services;
+    }
+
+    private static IConfiguration GetConfiguration()
+    {
+        var environment = Environment.GetEnvironmentVariable(AspnetcoreEnvironment);
+        var builder = new ConfigurationBuilder()
+            .AddJsonFile("appsettings.json", optional: false)
+            .AddJsonFile($"appsettings.{environment}.json", optional: true);
+        return builder.Build();
+    }
+
+    private static IServiceCollection RegisterLogging(IServiceCollection services, ITelemetryChannel telemetryChannel)
+    {
+        services.Configure<TelemetryConfiguration>(
+            config =>
+            {
+                config.ConnectionString = Environment.GetEnvironmentVariable(ApplicationInsightsConnectionString);
+                config.TelemetryChannel = telemetryChannel;
+            }
+        );
+
+        services.AddLogging(builder =>
+        {
+            builder.AddApplicationInsights();
+            // Console logging will be useful if we're investigating Console logs of a single job run
+            builder.AddConsole();
+        });
+        return services;
+    }
+}
+
+public static class ConfigurationExtension
+{
+    public static string GetRequiredValue(this IConfiguration config, string key) =>
+        config[key] ?? throw new ArgumentException($"{key} missing from the configuration / environment settings");
+}
+
+// BuildAssetRegistryContext needs an IHostEnvironment, but I'm pretty sure it's not used at all.
```

Comment:
I'd say this is wrong. I think we could still create an app host builder (of some sort, there's more)

Reply:
If we created everything using builder, it would automatically read the env variables and load appsettings.json correctly

Reply:
Using a HostApplicationBuilder now